### PR TITLE
nrf: cleanup /include directory

### PR DESCRIPTION
**Pull request by lemrey**
The `/include` directory could benefit from a little clean-up.
I am attempting to group public headers where it makes sense, so that they are more organized.
I will pile up several commits one at a time, so that CI runs on each of them.

One obvious category would be `/include/lte`, since a lot of libraries are actually LTE-specific e.g. all the AT libraries. `modem_` libraries would go in here too e.g. `/include/lte/modem_info.h`.

```
$ tree lte
lte
├── at_cmd.h
├── at_cmd_parser.h
├── at_notif.h
├── at_params.h
├── lte_lc.h
├── modem_info.h
├── modem_key_mgmt.h
├── pdn_management.h
└── sms.h
```

This looks fine, with the exception of `lte_lc.h` perhaps. I would choose a more descriptive name for `lte_lc.h` to avoid the repetition of `lte` in the include directive: `#include <lte/lte_lc.h>` for example `#include <lte/link_control.h>` or simply `#include <lte/link.h>`. The API could follow suit to become more descriptive: `lte_lc_` -> `lte_link_*` and nicely aligned with the new path; cc @jtguggedal.

Another category could be `/include/bootloader`. Here I am looking at grouping `bl_crypto.h` and `bl_validation.h` and possibly `fw_info.h` and `fprotect.h` e.g:
```
bootloader/
├── bl_crypto.h
├── bl_validation.h
├── fw_info.h
└── fprotect.h
```
Alternatively `/include/boot/` seems like a valid, shorter option.
I am not 100% satified with `#include <bootloader/bl_crypto.h`, since `bl` is in itself a short version of `bootloader`. perhaps it could be simplified? cc @hakonfam, @oyvindronningstad.

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
> I am not 100% satified with `#include <bootloader/bl_crypto.h`, since `bl` is in itself a short version of `bootloader`. perhaps it could be simplified? cc @hakonfam, @oyvindronningstad.

I agree. I say go ahead and remove the bl_ prefix.

**Comment by rlubos** [comment]
Huge thumbs up for keeping the lte-specific stuff grouped together 👍 Probably we should also consider moving `bsdlib.h` header from `include/net` into `/include/lte`. Plus there's `zzhc` documentation inside `include/net` as well - this could also be a part of the cleanup.

**Comment by jtguggedal** [comment]
@lemrey I agree on putting link controller into `lte` folder.
I've been thinking a bit about the link controller name and its functionality as well. At the moment what it does is more like a modem control module, and not limited to LTE. Also, it overlaps a bit with the `modem_info` module. IMO they're candidates for merging into a `modem` module.
I'm not saying we should do anything about it in this PR, though, just putting it out there.

**Comment by lemrey** [comment]
Rebased to fix docs.

**Comment by lemrey** [comment]
Any opinions on whether it should be `/include/lte` or `/include/modem` ?

**Comment by jtguggedal** [comment]
@lemrey I think `modem` makes most sense when we also keep `at_cmd` and other AT stuff in that folder, as that's not restricted to LTE only.
@jhn-nordic Any thoughts on this?

**Comment by lemrey** [comment]
I agree @jtguggedal; I think `/include/modem` would be more appropriate, being slightly more generic too. What I don't like too much are what some paths would look like:
- `#include <modem/modem_info.h>`
- `#include <modem/modem_key_mgmt.h>`

because of the repetition of `modem` e.g.  `<modem/modem_`

**Comment by lemrey** [comment]
@rlubos yep, I think `bsdlib` could be moved to `/modem/bsdlib.h` from `/net/bsdlib.h` 👍
Regarding `zzhc.rst`, where should I move that? That lib doesn't seem to have a public header either..

@jtguggedal I plan to move `nrf_cloud.h` to `/net/nrf_cloud.h`; then we would have three "cloud" headers there:
```
- nrf_cloud.h
- cloud.h
- cloud_backend.h
```
There isn't much in `cloud_backend.h`, perhaps I could merge it with `cloud.h`?

**Comment by rlubos** [comment]
> Regarding zzhc.rst, where should I move that? That lib doesn't seem to have a public header either..

Why did we put this in the `include/` instead of `doc/` section then 😕 Probably some mistake? @greg-fer I see you've reviewed the original PR, anything comes to your mind?

**Comment by umapraseeda** [comment]
> > Regarding zzhc.rst, where should I move that? That lib doesn't seem to have a public header either..
> 
> Why did we put this in the `include/` instead of `doc/` section then 😕 Probably some mistake? @greg-fer I see you've reviewed the original PR, anything comes to your mind?

Moving this file to doc\nrf\lib @ru-fu @greg-fer . 
@rlubos Can we go ahead with the current category under which this library will be displayed in the documentation (under libraries for networking: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/lib_networking.html )? Any thoughts on this?

**Comment by rlubos** [comment]
IMO just as we group modem-related headers, it would be good to have a section in the documentation for modem-related libraries. Currently, most of them ended up in "Other libraries" section.

**Comment by metc** [comment]
> Any opinions on whether it should be `/include/lte` or `/include/modem` ?

LTE is the standard, it includes LTE Cat M1 and LTE Cat NB1 for instance. IMO `lte` could be a subfolder in `modem` if needed, but a single modem folder LGTM.

This changes makes it easier to see which module can only be used on nRF9 series. This was not the case since then. The only way to use look at the KConfig.

As @rlubos mentioned it would be good to group the libraries made for nRF9 series in the documentation and also make sure then cannot be used without bsdlib on other series.

**Comment by lemrey** [comment]
Had to rebase to fix CI; will a couple more commits to move `nrf_cloud.h` and `bsdlib.h` into `/net` and `/modem` respectively. I will move the header files belonging to the bootloader in a separate PR to try and get this one merged soon-ish.

File: include/net/cloud.h [hunk 1]
```diff
@@ -374,6 +374,22 @@ static inline int cloud_user_data_set(struct cloud_backend *const backend,
...
 
+/**
+ * @brief Calls the user-provided event handler with event data.
+ *
+ * @param backend 	Pointer to cloud backend.
+ * @param evt		Pointer to event data.
+ * @param user_data	Pointer to user-defined data.
+ */
+static inline void cloud_notify_event(struct cloud_backend *backend,
+			struct cloud_event *evt,
```
Comment (rlubos) [inline include/net/cloud.h]: Alignment?

**Comment by lemrey** [comment]
Added changelog entry.

**Pull request merged**